Lib/ntpath.py

```diff
@@ -146,17 +146,28 @@ def splitdrive(p):
             sep = b'\\'
             altsep = b'/'
             colon = b':'
+            dev_prefix = b'\\\\?\\'
+            unc_prefix = b'UNC\\'
```

Comment:
Since you're only special casing the "?" (extended filesystem) namespace, and not the "." (device) namespace, I'd just call it `prefix`. 

With "?" restricted to the filesystem namespace, there are three path types to support: "//?/X:/" (drive letter), "//?/UNC/server/share/" (UNC), and "//?/Volume{12345678-1234-1234-1234-123456789ABC}/" (GUID). You'll need to explicitly support GUID paths. The existing code supports a GUID path as a generic UNC path with server "?" and share "Volume{GUID}".